src/openrct2/ride/Vehicle.cpp

```diff
@@ -7296,80 +7326,80 @@ static void sub_6D63D4(rct_vehicle * vehicle)
             vehicle_invalidate(vehicle);
         }
         break;
-    case 2: // loc_6D6424
+    case VEHICLE_ENTRY_ADDITIONAL_ANIMATION_2: // loc_6D6424
         *var_C8 += _vehicleVelocityF64E08;
         al = (*var_C8 >> 18) & 2;
-        if (vehicle->var_C5 != al)
+        if (vehicle->animation_frame != al)
         {
-            vehicle->var_C5 = al;
+            vehicle->animation_frame = al;
             vehicle_invalidate(vehicle);
         }
         break;
-    case 3: // loc_6D6482
+    case VEHICLE_ENTRY_ADDITIONAL_ANIMATION_3: // loc_6D6482
```

Comment:
type 5 is helicar, animation type 3 is rowboats
type 8 is monorail cycles, type 6 is the observation tower
type 9 is 4D coaster, and I don't know about 2, 4, or 7